### PR TITLE
Create app checks | Removed search | Remove custom app from dropdown

### PR DESCRIPTION
**Pull request by milindk8**
# Description

1. Remove custom app from dropdown until custom app config checklist is complete in full mode
2. EAOnly > [Create new] missing for new chart deployment
3. Primary Nav > Comment out "Search"

Fixes - https://github.com/devtron-labs/devtron/issues/1724

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
All of the tests are done manually

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/list-new/AppList.tsx [hunk 1]
```diff
@@ -40,9 +40,9 @@ export default function AppList() {
     const [showCreateNewAppSelectionModal, setShowCreateNewAppSelectionModal] = useState(false)
 
     // API master data
-    const [appCheckListRes, setAppCheckListRes] = useState({})
-    const [projectListRes, setProjectListRes] = useState({ result: [] })
-    const [environmentListRes, setEnvironmentListRes] = useState({ result: [] })
+    const [appCheckListRes, setAppCheckListRes] = useState({result: undefined});
```
Comment (vivek-devtron) [inline src/components/app/list-new/AppList.tsx:43]: We should not set 'undefined' manually

File: src/components/app/list-new/AppList.tsx [hunk 2]
```diff
@@ -871,15 +871,17 @@ export default function AppList() {
                 rootClassName="app-create-model-wrapper"
                 onClick={() => setShowCreateNewAppSelectionModal(!showCreateNewAppSelectionModal)}
             >
-                <div className="app-create-child c-pointer" onClick={openDevtronAppCreateModel}>
-                    <AddIcon className="icon-dim-20 fcn-9" />
-                    <div className="ml-8">
-                        <strong>Custom app</strong>
-                        <div>
-                            Connect a git repository to deploy <br /> a custom application
+                {serverMode == SERVER_MODE.FULL && appCheckListRes?.result?.appChecklist && Object.values(appCheckListRes.result.appChecklist).every((check) => check)  && (
```
Comment (vivek-devtron) [inline src/components/app/list-new/AppList.tsx:874]: on line no 43 we are initializing appCheckListRes so no need to '?' after appCheckListRes

File: src/components/common/navigation/Navigation.tsx [hunk 3]
```diff
@@ -25,14 +25,14 @@ import ReactGA from 'react-ga'
...
 const NavigationList = [
-    {
-        title: 'Search (⌘+/)',
-        type: 'button',
-        iconClass: 'nav-short-search',
-        icon: SearchIcon,
-        href: URLS.APP,
-        isAvailableInEA: true,
-    },
+    // {
```
Comment (vivek-devtron) [inline src/components/common/navigation/Navigation.tsx:28]: Please remove commented code

File: src/components/app/list-new/AppList.tsx (hunk 2 above)
Comment (milindk8) [reply src/components/app/list-new/AppList.tsx]: done

File: src/components/common/navigation/Navigation.tsx (hunk 3 above)
Comment (milindk8) [reply src/components/common/navigation/Navigation.tsx]: removed

**Review by vivek-devtron — APPROVED** [review]
Should also remove all the component and resource related to search

**Comment by sohel-rp** [comment]
> Should also remove all the component and resource related to search

Yup, if we're not going to add it back. 
Let's keep it for some time and then clean-out related files, if not required anymore.

File: src/components/app/list-new/AppList.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/app/list-new/AppList.tsx]: It's been replaced with `null` for now. However, we should be having some specific type for it.

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**